src/redux/slices/postFunctions.js

```diff
@@ -0,0 +1,100 @@
+import { createAsyncThunk } from "@reduxjs/toolkit";
+import { updateDoc, arrayUnion, doc, arrayRemove, increment } from "@firebase/firestore";
+import { db } from "../../firebase";
+
+export const addBookmark = createAsyncThunk(
+    "post/addBookmark", async (arg) => {
+        try {
+            const userRef = doc(db, `users/${arg.userID}`);
+            const res = await updateDoc(userRef, {
+                bookmarks: arrayUnion(arg.id)
+            });
+            const localUser = JSON.parse(localStorage.getItem("user"));
+            localStorage.setItem("user", JSON.stringify({ ...localUser, bookmarks: [...localUser.bookmarks, arg.id] }));
```

Comment:
Can have a check on the `localUser` variable whether it is fetched from the local storage successfully ]JSON.parse sometimes returns null or undefined if the variable doesn't exist in it] or not then only set the item in the local Storage in case of potential errors !!